lib/SelectedObjectsProvider/useObjectSelector.js

```diff
@@ -0,0 +1,49 @@
+import { useContext } from 'react';
+import { SelectedObjectsContext } from './index';
+
+export function useObjectSelector(
+  id,
+  type,
+  idsToSelect,
+  disabledCallback = () => false
+) {
+  const {
+    selected,
+    selectMultiple,
+    deselectMultiple,
+    updateSelected,
+    currentSelectedType,
+    intendedToSelect,
+    setIntendedToSelect
+  } = useContext(SelectedObjectsContext);
+
+  const isSelected = selected.indexOf(id) !== -1;
+  const isDisabled = disabledCallback(currentSelectedType, isSelected);
+  const isHovered = intendedToSelect.indexOf(id) !== -1;
+
+  const getIds = (additionalIds = []) => [
+    ...new Set([id, ...idsToSelect, ...additionalIds])
+  ];
+
+  const handleClick =
+    idsToSelect.length > 1
+      ? () =>
+          isSelected
+            ? deselectMultiple(getIds(), type)
+            : selectMultiple(getIds(), type)
+      : () => updateSelected(idsToSelect[0]);
+
+  const handleMouseEnter = () => setIntendedToSelect(idsToSelect);
+
+  const handleMouseLeave = () =>
+    setIntendedToSelect(intendedToSelect.filter(s => !idsToSelect.includes(s)));
```

Comment:
Unfortunately includes isn't supported by IE (damn you IE 😾).

you could check the transpiled code to see if we have a polyfill for this, but I'm pretty sure last time I checked we didn't.

If we do, you could also replace the uses of `array.indexOf(id) !== -1`, which is our workaround for not being able to use `includes`.